tests/testthat/test-derive_vars_dt.R

```diff
@@ -1,105 +1,140 @@
-# impute_dtc_dt ----
 input <- c(
-  "2019-07-18",
-  "2019-02",
-  "2019",
-  "2019---07"
+  "2019-07-18", # full date
```

Comment:
@bms63 @millerg23 @StefanThoma before I proceed further down the function/more functions, this is roughly the thing I had in mind, but then in occurred to me that we need to suppress warnings a lot, any thoughts?

Reply:
Well it's more for the date-ish looking ones we expected a matched output, but where we expect things to go wrong, we want the warnings, but the value should map over to `NA`

Reply:
yeah - maybe a bad data/extreme example would be nice to test things separately?

Reply:
Makes more sense that way, will do